### DISCUSSION TITLE
How to set hdf5 include directory?

### DISCUSSION THREAD

**Discussion by Fabian188** [CGNS/CGNS#363] [CGNStalk]
We build hdf5 by ourselves. Currently this is 1.8.20

I set our hdf5-lib via  HDF5_C_LIBRARY_hdf5, but how can I point to our include? cmake uses the system hdf5-include.

When I set the lib to our 1.8.20 lib, I get the error

```
Consolidate compiler generated dependencies of target cgns_static
[ 24%] Built target cgns_static
Consolidate compiler generated dependencies of target cgns_shared
[ 48%] Built target cgns_shared
[ 56%] Built target cgnsnames
[ 59%] Linking C executable cgnscheck
/usr/bin/ld: ../libcgns.so.4.3: undefined reference to `H5Oget_info_by_name1'
collect2: error: ld returned 1 exit status
```
I assume this is because the wrong system hdf5 include is used, and the renaming macro handled wrong
https://docs.hdfgroup.org/hdf5/v1_12/group___h5_o.html

Anyway, the lib builds, it's just the check which makes the problems which I don't need. I open another discussion

**Comment by mennodeij** [comment]
You should I think, give `HDF5_DIR` as parameter to CMake as the directory where the hdf5-config.cmake file is located:

`-DHDF5_DIR=<path/to/your/hdf5-install>/share/cmake`

**Reply by Fabian188** [reply]
Thanks, I see. Currently we don't store the hdf5-config.cmake when we build hdf5. We have kind of precompiled cmake external projects. We keep the libs and includes and unpack wherever we build. Usually we just provide the lib and include for other libs.

I assume that you need hdf5-config.cmake also for the hdf5 configuration. 

I see two options:
* I provide an appropriate hdf5-config.cmake
* I provide all the information you need from hdf5-config.cmake from outside via -D

Could you judge if the second option is feasible? Thanks!

**Comment by mennodeij** [comment]
hdf5-config.cmake is a file produced by the HDF5 build that tells consumers of the library where the include files are, and what the targets (i.e. libraries) are that the dependent program needs to link to. All in all it is much easier to have the hdf5-config.cmake file.
